docs/remote-k8s-access/README.md

```diff
@@ -0,0 +1,179 @@
+
+This document describes what is involved with creating and managing connections to remote clusters.
+
+When the operator starts up it queries for ClientConfig objects. A ClientConfig provides the necessary information for the operator to create client connections to remote clusters. 
+
+Before getting into details of using ClientConfigs we need to provide some background on service accounts, kubeconfig files, and secrets.
+
+## ServiceAccount
+All pods are associated with a ServiceAccount. If no ServiceAccount is specified for a pod, then the default one will be used.
+
+**Note:** Kubernetes creates a default ServiceAccount in each namespace.
+
+The ServiceAccount is used to authenticate and authorizate the pod for any operations that it wants to perform against the api server. Authentication is done by way of a service account token which is stored in a secret. Kubernetes automatically mounts the secrect in the pod. Authorization is done with Roles and RoleBindings. The ServiceAccount is mapped to Roles and RoleBindings with RoleBindings and ClusterRoleBindings respectively.
+
+To learn more about this, see the official Kubernetes [docs](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod) for accessing the api server from a pod.
+
+A service account for the operator should be created in each cluster that the operator needs to access. This is done automatically as part of the operator installation.
+
+**Note:** The operator needs to be installed in each cluster to which it needs access.
+
+## Kubeconfig
+A kubeconfig file configures access to one or more Kubernetes clusters. Clients such as kubectl or the [client-go](https://github.com/kubernetes/client-go) library use kubeconfig files to access the api server.
+
+K8ssandra Operator uses kubeconfig files to create remote client connections.
+
+A `context` is a container object in a kubeconfig file that has three properties:
+
+* `cluster`
+* `user`
+* `namespace`
+
+`cluster` is a reference to a `cluster` object which declares the URL and the CA cert of the api server.
+
+`user` is a reference to a `user` object which specifies details to authenticate against the api server. This could for example include a client certificate and key or a bearer token.
+
+**Note:** A service account token is a bearer token.
+
+`namespace` is a default namespace to use for requests. This is optional.
+
+Here is an example kubeconfig file that uses a bearer token:
+
+```yaml
+apiVersion: v1
+clusters:
+- cluster:
+    certificate-authority-data: ...
+    server: https://127.0.0.1:65318
+  name: kind-k8ssandra-0
+contexts:
+- context:
+    cluster: kind-k8ssandra-0
+    user: kind-k8ssandra-0-k8ssandra-operator
+  name: kind-k8ssandra-0
+current-context: kind-k8ssandra-0
+kind: Config
+preferences: {}
+users:
+- name: kind-k8ssandra-0-k8ssandra-operator
+  user:
+    token: ...
+```
```

Comment:
A bit of reordering would make the file more naturally readable by a human:

```suggestion
```yaml
apiVersion: v1
kind: Config
preferences: {}
clusters:
  - name: kind-k8ssandra-0
    cluster:
      certificate-authority-data: ...
      server: https://127.0.0.1:65318 
users:
  - name: kind-k8ssandra-0-k8ssandra-operator
    user:
      token: ...
contexts:
  - name: kind-k8ssandra-0
    context:
      cluster: kind-k8ssandra-0
      user: kind-k8ssandra-0-k8ssandra-operator
current-context: kind-k8ssandra-0
```
```